### PR TITLE
Better error handing for images/create compat api

### PR DESCRIPTION
**Pull request by matejvasek**
If server cannot resolve short name due to enforcing policy send
non-2xx status code.

Signed-off-by: Matej Vasek <mvasek@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/compat/images.go [hunk 1]
```diff
@@ -231,6 +232,12 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 
 	fromImage := mergeNameAndTagOrDigest(query.FromImage, query.Tag)
 
+	_, err := shortnames.Resolve(runtime.SystemContext(), fromImage)
```
Comment (vrothberg) [inline pkg/api/handlers/compat/images.go:235]: Can you drop a comment and elaborate why do an early check here?

Future readers may feel inclined to drop the check.

**Review by vrothberg — COMMENTED** [review]
Code LGTM
@containers/podman-maintainers PTAL

File: pkg/api/handlers/compat/images.go [hunk 2]
```diff
@@ -231,6 +232,14 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 
 	fromImage := mergeNameAndTagOrDigest(query.FromImage, query.Tag)
 
+	// without this early check this function would return 200 but reported error via body stream soon after
+	// it's better to let caller know early via HTTP status code that request cannot be processed
+	_, err := shortnames.Resolve(runtime.SystemContext(), fromImage)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrap(err, "failed to resolve image name"))
```
Comment (matejvasek) [inline pkg/api/handlers/compat/images.go:239]: Maybe `403 Forbidden` would be better?

Comment (vrothberg) [reply pkg/api/handlers/compat/images.go:239]: That's @jwhonce question :) He's the Yoda of REST APIs.

Comment (matejvasek) [reply pkg/api/handlers/compat/images.go:239]: Thinking about it I cannot be sure that this failed due to enforcing mode. Maybe it should be internal server error after all.

**Comment by rhatdan** [comment]
/approve
/lgtm
Will allow @jwhonce Merge.

**Pull request merged**